internal/groups/api/endpoint.go

```diff
@@ -0,0 +1,369 @@
+package api
+
+import (
+	"context"
+
+	"github.com/go-kit/kit/endpoint"
+	"github.com/mainflux/mainflux/auth"
+)
+
+func createGroupEndpoint(svc auth.Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(createGroupReq)
+		if err := req.validate(); err != nil {
+			return groupRes{}, err
+		}
+
+		group := auth.Group{
```

Comment:
If we are moving Groups away from Auth microservice, why would the Group struct type stay defined there?

Reply:
Merging this one, and @dusanb94 will continue working on a branch and correct those. Thanks @buraksekili.